src/lib/__tests__/insights-client-test.ts

```diff
@@ -0,0 +1,261 @@
+import { withInsights, inferInsightsPayload } from '../insights';
+
+const connectHits = (renderFn: any, unmountFn: any) => (widgetParams = {}) => ({
+  init() {},
+  render({ results, instantSearchInstance }) {
+    const hits = results.hits;
+    renderFn({ hits, results, instantSearchInstance, widgetParams }, false);
+  },
+  dispose() {
+    unmountFn();
+  },
+});
+
+const setup = ({ hits: providedHits, insightsClient }: any = {}) => {
```

Comment:
I'm not convinced by this setup. There's a lot of logic and unnecessary abstraction that we should avoid in the tests.

Reply:
Please explain the solution you have in mind.

Reply:
I meant inlining this in the tests themself.

Reply:
I think that would be way too verbose. That's 20 lines to copy paste in every tests.

Reply:
Verbose sometimes means easier to reason about.

Here it creates so much abstraction with instructions that have very different purposes. Abstraction in tests is also error prone because the abstraction is not tested itself.

You create too much boilerplate that is not needed in many tests. Besides, some (future) tests might not behave correctly because of:

```js
const widgetParams = { transformItems: jest.fn() };
```

The tests are really hard to understand in isolation. There are too much side effects.

Honestly, who cares about having verbose tests? That's something we already have a lot in the codebase anyway.